supertokens_python/normalised_url_path.py

```diff
@@ -40,7 +40,9 @@ def equals(self, other: NormalisedURLPath) -> bool:
 
     def is_a_recipe_path(self) -> bool:
         parts = self.__value.split("/")
-        return parts[1] == "recipe" or parts[2] == "recipe"
+        return (len(parts) > 1 and parts[1] == "recipe") or (
+            len(parts) > 2 and parts[2] == "recipe"
```

Comment:
why this change in this PR?

Reply:
Tests try to query `/api`, `/api1`, `/api2/`, etc. They throw error if we don't do this. Better to just return True/False from this?